### PR TITLE
Handle single element cases in MerkleTree and update tests

### PR DESCRIPTION
**Pull request by jotabulacios**
## Description

This pull request introduces changes to ensure that the MerkleTree implementation correctly handles cases where there is only a single element. 
The changes include:

- Add a condition in the MerkleTree::build method to handle single element cases.

## Type of change

Please delete options that are not relevant.

- [x ] Bug fix

## Checklist

  - [x ] Benchmarks added/run

### CODE REVIEW

File: crypto/benches/criterion_merkle.rs [hunk 1]
```diff
@@ -8,29 +8,41 @@ use lambdaworks_math::{
...
 
 fn merkle_tree_benchmarks(c: &mut Criterion) {
     let mut group = c.benchmark_group("Merkle Tree");
     group.sample_size(10);
     group.measurement_time(Duration::from_secs(30));
 
-    // NOTE: the values to hash don't really matter, so let's go with the easy ones.
+    // Large number of leaves benchmark
     let unhashed_leaves: Vec<_> = core::iter::successors(Some(FE::zero()), |s| Some(s + FE::one()))
-        // `(1 << 20) + 1` exploits worst cases in terms of rounding up to powers of 2.
```
Comment (Oppen) [inline crypto/benches/criterion_merkle.rs:23]: The comments explaining the benchmark were there for a reason. Is there a reason to remove them?

File: crypto/src/hash/sha3/mod.rs [hunk 2]
```diff
@@ -6,8 +6,6 @@ use sha3::{Digest, Sha3_256};
 
 pub struct Sha3Hasher;
 
-/// Sha3 Hasher used over fields
-/// Notice while it's generic over F, it's only generates enough randomness for fields of at most 256 bits
```
Comment (Oppen) [inline crypto/src/hash/sha3/mod.rs:10]: Yet again, do we need to remove explanatory comments here?

**Review by entropidelic — APPROVED** [review]

File: crypto/src/merkle_tree/merkle.rs [hunk 3]
```diff
@@ -125,9 +122,12 @@ mod tests {
         const MODULUS: u64 = 13;
         type U64PF = U64PrimeField<MODULUS>;
         type FE = FieldElement<U64PF>;
-
-        let values: Vec<FE> = vec![FE::new(1)];
+
```
Comment (Oppen) [inline crypto/src/merkle_tree/merkle.rs]:
```suggestion

```

File: crypto/src/merkle_tree/merkle.rs [hunk 4]
```diff
@@ -8,7 +8,7 @@ use super::{proof::Proof, traits::IsMerkleTreeBackend, utils::*};
 pub enum Error {
     OutOfBounds,
 }
-
+
```
Comment (Oppen) [inline crypto/src/merkle_tree/merkle.rs]:
```suggestion

```

File: crypto/src/merkle_tree/merkle.rs [hunk 5]
```diff
@@ -36,10 +36,7 @@ where
 
         // If there is only one node, handle it specially
         if hashed_leaves.len() == 1 {
-            return MerkleTree {
-                root: hashed_leaves[0].clone(),
-                nodes: hashed_leaves,
-            };
+            hashed_leaves.push(hashed_leaves[0].clone());
```
Comment (Oppen) [inline crypto/src/merkle_tree/merkle.rs:39]: I'm not sure I understand this change. Why does it no longer return?

**Review by Oppen — APPROVED** [review]

File: crypto/src/merkle_tree/proof.rs [hunk 6]
```diff
@@ -152,4 +152,29 @@ mod tests {
             assert_eq!(node, expected_node);
         }
     }
+
+    #[test]
+    fn build_merkle_tree_from_a_single_value() {
```
Comment (entropidelic) [inline crypto/src/merkle_tree/proof.rs:157]: a better name would be `fn verify_merkle_proof_for_single_value()`

**Pull request merged**